metric/system/process/process_windows.go

```diff
@@ -463,3 +468,26 @@ func fillIdleProcess(state ProcState) (ProcState, error) {
 	state.CPU.Total.Value = opt.FloatWith(idle)
 	return state, nil
 }
+
+func shouldIgnore(pid int) bool {
+	// shouldIgnore checks if we should ignore the pid, to avoid elevated permissions
+
+	// LSASS.exe is a process which has no useful cmdline arguments, we should ignore acessing such process to avoid triggering Windows ASR rules
+	// we can query pid for LASASS.exe from registry
+
+	key, err := registry.OpenKey(registry.LOCAL_MACHINE, "SYSTEM\\CurrentControlSet\\Control\\Lsa", registry.READ)
```

Comment:
Are there any performance concerns doing this unconditionally for every PID in the system?

This could probably happen once right before iterating over every PID and have a low chance of being wrong, unless we know this PID changes all the time.

https://github.com/elastic/elastic-agent-system-metrics/blob/5a3873d8d6f807e2c111b437c10291a13d9c2b83/metric/system/process/process_windows.go#L51-L57

Reply:
@cmacknz yeah, I think moving this to global would solve this. PIDs of such process shouldn't change.

Reply:
> yeah, I think moving this to global would solve this. PIDs of such process shouldn't change.

@gabriellandau Can the lssas.exe PID change without a full system restart? Is it safe to read once and cache the value.

Reply:
From https://en.wikipedia.org/wiki/Local_Security_Authority_Subsystem_Service,

> Forcible termination of lsass.exe will result in the system losing access to any account, including NT AUTHORITY, prompting a restart of the machine

Many articles mentioned similar thing i.e. if lsass.exe shut's down, then system is rebooted. 

@gabriellandau what do you think?

Reply:
> @gabriellandau Can the lssas.exe PID change without a full system restart?

Negative.  One per boot.  It's [safe to cache](https://github.com/elastic/beats/issues/41407#issue-2609676190).

> Its PID can be found in the registry. We can query that value once and cache the value because it will never change until reboot.